src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/AppleCertificatePal.Pkcs12.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography.Apple;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed partial class AppleCertificatePal : ICertificatePal
+    {
+        private static ICertificatePal ImportPkcs12(
+            byte[] rawData,
+            SafePasswordHandle password,
+            bool exportable,
+            SafeKeychainHandle keychain)
+        {
+            using (ApplePkcs12Reader reader = new ApplePkcs12Reader(rawData))
+            {
+                reader.Decrypt(password);
+
+                UnixPkcs12Reader.CertAndKey certAndKey = reader.GetSingleCert();
+                AppleCertificatePal pal = (AppleCertificatePal)certAndKey.Cert;
+
+                SafeSecKeyRefHandle safeSecKeyRefHandle =
+                    ApplePkcs12Reader.GetPrivateKey(certAndKey.Key);
+
+                AppleCertificatePal newPal;
+
+                using (safeSecKeyRefHandle)
+                {
+                    // SecItemImport doesn't seem to respect non-exportable import for PKCS#8,
+                    // only PKCS#12.
+                    //
+                    // So, as part of reading this PKCS#12 we now need to write the minimum
+                    // PKCS#12 in a normalized form, and ask the OS to import it.
+                    if (!exportable && safeSecKeyRefHandle != null)
+                    {
+                        return ImportPkcs12NonExportable(pal, safeSecKeyRefHandle, password, keychain);
+                    }
+
+                    newPal = pal.MoveToKeychain(keychain, safeSecKeyRefHandle);
```

Comment:
considering the null check above should this also be guarded against null handle?

Reply:
MoveToKeychain handles a null safeSecKeyRefHandle.

The null check is only on the previous one for "if we're not exportable and we have a private key", if there's no private key we don't need to do the export/import handshake.  If it's exportable we also don't need to do it.